source/Group7_GP_Agent/LGPAgent.hpp

```diff
@@ -0,0 +1,174 @@
+
+#pragma once
+
+#include "../core/AgentBase.hpp"
+#include "GPAgentSensors.hpp"
+#include <algorithm>
+#include <iostream>
+#include <random>
+#include <string>
+#include <vector>
+
+const int LISTSIZE = 100;
+
+/// Generated by CHATGPT
+
+
+#ifndef NDEBUG
+#define EXPERIMENTAL_CLASS                                                     \
+  __attribute__((                                                              \
+      annotate("experimental_class"),                                          \
+      warning(                                                                 \
+          "This is an experimental class and should be used with caution.")))
+#define EXPERIMENTAL_FUNCTION                                                  \
+  __attribute__((annotate("experimental_function"),                            \
+                 warning("This is an experimental function and should be "     \
+                         "used with caution.")))
+
+#else
+
+#define EXPERIMENTAL_CLASS                                                     \
+  __attribute__((                                                              \
+      annotate("experimental_class"),                                          \
+      error(                                                                 \
+          "This is an experimental class and is currently being worked on and should not be used in release")))
+#define EXPERIMENTAL_FUNCTION                                                  \
+  __attribute__((annotate("experimental_function"),                            \
+                 error("This is an experimental function and is currently being worked on and should not be used in release")))
+#endif
+
+
+
+
+namespace cowboys {
+
+EXPERIMENTAL_CLASS class LGPAgent : public cse491::AgentBase {
+protected:
+  // A dictionary of actions and a dictionary of sensors
+  // A sensor is a function that takes in a grid and returns a value (e.g.
+  // distance to nearest agent)
+
+  // For example group 1 has a function for the shortest path
+  // TODO: ADD A STAY STATEMENT WORLD
+
+  std::vector<std::string> possibleInstructionsList = {};
+  std::vector<std::string> actionsList = {"up", "down", "left", "right"};
+  std::vector<std::string> operationsList = {"lessthan", "greaterthan",
+                                             "equals"};
+  std::vector<int> resultsList = std::vector<int>(LISTSIZE);
+
+  std::vector<std::tuple<std::string, int, int>> instructionsList = {};
+  size_t currentInstructionIndex = 0;
+
+  std::vector<cowboys::Sensors> sensorsList;
+  std::vector<std::string> sensorsNamesList = {"getLeft", "getRight", "getUp",
+                                               "getDown"};
+
+  bool hasActionOccured = false;
+
+  std::random_device rd;
```

Comment:
This brings up an important point -- should we have a single random number generator for the whole project that gets passed around?  That would allow a seed to be set and make the run perfectly repeatable (assuming a human, if any, takes the exact same actions).

Reply:
@Rajmeet @JasonMSeguin

Reply:
If we have a shared RNG, would using the RNG in one place affect the random numbers that we get in another place? Probably doesn't matter that much in the end, but if something changes and calls the RNG an extra time, wouldn't the outcome from the same seed will completely change? Then again if something changes like that it would probably change everything anyway.